src/base/net/downloadmanager.h

```diff
@@ -42,16 +46,40 @@ namespace Net
 {
     class DownloadHandler;
 
+    struct DownloadRequest
+    {
+        DownloadRequest(const QString &url);
+
+        QString url;
+        bool saveToFile = false;
+        qint64 limit = 0;
+        bool handleRedirectToMagnet = false;
+        QString userAgent;
+    };
+
+    struct ServiceID
```

Comment:
>This is certainly not and "identifier". Following asio this is an endpoint.

The "identifier" is a fairly broad concept. It's just some unique information that allows you to distinguish some objects. So the endpoint is also an identifier. But I can use the name "endpoint" if you want.
>Also it is not clear why the class is needed, i.e. why it is assumed that server subscription settings depend only on these two parameters.

I don't understand what you mean by "server subscription."

As a rule, some Internet service (or rather, some instance) is uniquely determined by the host name and port. I need to prevent parallel requests from being sent to some services (specifically websites from which users receive RSS feeds).